robotbase/src/main/java/frc/robot/commands/intake/IntakeNoteFromFloor.java

```diff
@@ -1,44 +1,19 @@
 package frc.robot.commands.intake;
 
-import edu.wpi.first.wpilibj2.command.Command;
-import frc.robot.Constants.*;
-import frc.robot.Robot;
-import frc.robot.commands.rumble.RumbleDriverController;
-import frc.robot.state.RobotState;
-
-public class IntakeNoteFromFloor extends Command {
-
-  private Command m_rumbleDriverController;
+import edu.wpi.first.wpilibj2.command.SequentialCommandGroup;
+import frc.robot.Constants.INTAKE;
+import frc.robot.commands.state.SetIntakeState;
+import frc.robot.state.RobotState.IntakeState;
 
+public class IntakeNoteFromFloor extends SequentialCommandGroup {
   public IntakeNoteFromFloor() {
-    addRequirements(Robot.intake);
-    m_rumbleDriverController = new RumbleDriverController();
-  }
-
-  @Override
-  public void execute() {
-    Robot.intake.set(
-        Robot.intake.isBeamBroken()
-            ? INTAKE.TOP_MOTOR_SLOW_PICKUP_SPEED_PERCENT_OUTPUT
-            : INTAKE.TOP_MOTOR_PICKUP_SPEED_PERCENT_OUTPUT,
-        Robot.intake.isBeamBroken()
-            ? INTAKE.BOTTOM_MOTOR_SLOW_PICKUP_SPEED_PERCENT_OUTPUT
-            : INTAKE.BOTTOM_MOTOR_PICKUP_SPEED_PERCENT_OUTPUT);
-  }
-
-  @Override
-  public boolean isFinished() {
-    return Robot.intake.hasNotePassedIntake();
-  }
-
-  @Override
-  public void end(boolean interrupted) {
-    Robot.intake.stop();
-    Robot.intake.resetNotePassedBeamBreak();
-
-    if (!interrupted) {
-      Robot.state.setState(RobotState.State.NOTE_STOWED);
-      m_rumbleDriverController.schedule();
-    }
+    super(
+        new IntakeRunUntilBeamState(INTAKE.PICKUP_SPEED_PERCENT_OUTPUT, true),
+        new SetIntakeState(IntakeState.NOTE_IN_INTAKE),
+        new IntakeRunUntilBeamState(INTAKE.SLOW_PICKUP_SPEED_PERCENT_OUTPUT, false),
+        new SetIntakeState(IntakeState.NOTE_PAST_BEAM_BREAK),
+        new IntakeAxis(() -> INTAKE.REVERSE_FEED_SPEED_PERCENT_OUTPUT)
```

Comment:
Should use a more generic set speed command instead of IntakeAxis in case we are doing something like deadbanding or curving in the setAxisSpeed method on the intake